React Capstone Project

rmiyoyo opened the pull request:
# Work Done
- When the page loads, the webapp shows the list of categories that could be filtered out by some parameter, for example by the category name.
- In the details page, the webapp retrieves data from the API to show detailed data in the category.
- When the user clicks on the "Back" button (<), the user navigates to the home page.
- Ran linter checks

rmiyoyo committed:
Deploy app

rmiyoyo committed:
Add routes

rmiyoyo committed:
Add path

rmiyoyo committed:
Deploy

rmiyoyo committed:
Add .env

rmiyoyo committed:
Adjust lat, lang

rmiyoyo committed:
Update packages

rmiyoyo committed:
Add path

rmiyoyo committed:
Add path

rmiyoyo committed:
Remove s from http

rmiyoyo committed:
Add GH Pages and Presentation

rmiyoyo committed:
Update packages

rmiyoyo committed:
fix live deployment

Nemwel-Boniface commented on README.md:
```diff
@@ -70,7 +70,13 @@
+## 🚀 Presentation <a name="presentation"></a>
+
+- [Presentation] (https://drive.google.com/file/d/1pauRlxyAOAUf61stpsAPvNjkTXbzmcSy/view?usp=share_link)
 
```
- [ ] I was not able to view the presentation video since I do not have access. Kindly remove access restrictions.

![image](https://github.com/rmiyoyo/metrics-webapp/assets/86318284/f9ce16e9-01ad-41e3-ab95-a586c8dd6404)

Nemwel-Boniface commented on README.md:
```diff
@@ -70,7 +70,13 @@
+
+- [Presentation] (https://drive.google.com/file/d/1pauRlxyAOAUf61stpsAPvNjkTXbzmcSy/view?usp=share_link)
 
 <p align="right">(<a href="#readme-top">back to top</a>)</p>
```
**As I am not able to access the Acknowledgements section, I have to highlight here.**

- [ ] As mentioned in the [project requirements](https://github.com/microverseinc/curriculum-react-redux/blob/main/capstone/react_capstone.md), kindly ensure that you give proper credit to the original designer ensuring that you include both the name of the designer as well as the link to the original design.

![image](https://github.com/rmiyoyo/metrics-webapp/assets/86318284/a6fdf148-f636-4045-a99a-32c04398a70d)

- [ ] KIndly change the title of the README which is currently titled as "Meals Project " while the project is not about meals.

![image](https://github.com/rmiyoyo/metrics-webapp/assets/86318284/6c5bd6f9-4d8a-4916-be61-8dc75ecbe0e3)

Nemwel-Boniface commented on src/App.js:
```diff
@@ -19,7 +19,7 @@ function App() {
-        <Route path="/" element={<Layout />}>
+        <Route path="/metrics-webapp" element={<Layout />}>
           <Route index element={<Home />} />
           <Route path="details" element={<Info />} />
```
- [ ] When you navigate to the info page, there is no way to navigate back to the home page as required by the provided design. Your design seems to be missing a back button which leads to the home/ root page.
- [ ] Also for the section with "Other pollutant materials" kindly note that the list is supposed to be separated with a space in between them.

**Below is your design:**

![image](https://github.com/rmiyoyo/metrics-webapp/assets/86318284/35894899-9a8c-4d8e-9a9a-d4ead6ff1caa)

**This is the provided design:**

![image](https://github.com/rmiyoyo/metrics-webapp/assets/86318284/e24a458e-e204-4990-afba-02d24ec1fc19)

Nemwel-Boniface commented on src/tests/Townlist.test.jsx:
```diff
@@ -1,10 +1,9 @@
 import React from 'react';
```
- [ ] The currently written tests seem to be failing. KIndly have a look and resolve the issue.

![image](https://github.com/rmiyoyo/metrics-webapp/assets/86318284/436e6440-435f-4936-a46a-6c0a86f2d4f1)

- [ ] I have realized that there are no tests for the Info page. KIndly include more tests testing the info page.
- [ ] I also noticed that the provided tests primarily focus on testing the React components that use Redux. While it tests the rendering and interaction of the components, it doesn't directly test the Redux logic in isolation. For example, the third test case checks whether a dispatched action occurs when a specific event (a click on a town item) is simulated. This indirectly verifies that the component interacts with Redux correctly when an action is triggered.Therefore, your written tests do not fulfil all the project requirements for writing tests "Create unit tests for pure functions in your Redux code." Kindly ensure that you include tests that test the pure functions in your Redux code as mentioned in the project requirements..

![image](https://github.com/rmiyoyo/metrics-webapp/assets/86318284/c60efe8e-0925-4b7c-9ec3-13d0ca6fadd2)

Nemwel-Boniface reviewed (changes requested):
# Hello @rmiyoyo  :wave: 

### This is a manual review

# :recycle:  Request Changes :recycle: 

Good job working on your project :clap: However, there are a few issues that need to be addressed to move to the next milestone.

![](https://media.giphy.com/media/KyBesf8kNT6s88SEgb/giphy.gif)

**Because some of the files cannot be highlighted, I will include them here**

- [ ] For some reason when I try to start the project locally, I keep getting this error and I was not able to test the project locally. It however worked when I checked the live demo.

![image](https://github.com/rmiyoyo/metrics-webapp/assets/86318284/8ea427c9-79e9-4251-9c67-97defcb6edbd)



- [ ] Kindly have a look at the inline comments left below.

Cheers, and Happy coding! :rocket:

rmiyoyo committed:
Change presentation link, update repo name, add acknowledgment

rmiyoyo committed:
Add back arrow to weather cards

rmiyoyo committed:
Add spacing between pollutants

rmiyoyo committed:
Fix styles linter errors

rmiyoyo committed:
Add tests

rmiyoyo committed:
Update packages

rmiyoyo committed:
Change https to http

rmiyoyo committed:
Fix eslint linter errors

rmiyoyo commented:
Hello @Nemwel-Boniface thank you for reviewing my code. I have implemented all the changes you requested.
# Work Done After Review
- Changed permissions on my presentation link.
- Added credits to the original designer in my acknowledgement section.
- Fixed the project name in the Readme.
- Added spaces between the pollutants list.
- Added more tests.
- Added back arrow to the cards
- Changed the links from https to http. Not sure if this was the cause for the project not running locally for you, if not, I would assume it is because the link has to have metrics-webapp(http://localhost:3000/metrics-webapp)

bellom commented:
Hi @rmiyoyo Good job so far! 👍 

### This is a manual review

♻️ Request Changes ♻️

Good knowledge hiding your API key, however, you might need to share it with us so we can review your project accordingly.

```
const ipgeoApiKey = process.env.REACT_APP_IPGEO_API_KEY;
const openweatherApiKey = process.env.REACT_APP_OPENWEATHER_API_KEY;
```

Also, kindly double-check your live demo link, seems not fetching data from the API anymore - https://rmiyoyo.github.io/metrics-webapp/

Kindly reply on this thread, and tag my name @bellom.

Thank you.

rmiyoyo committed:
Add API keys to homeSlice

rmiyoyo commented:
Hi @bellom, thank you so much for getting back! I have just added the keys to the project.
Also, the live demo is running now. Let me know if you need anything, thanks!

https://github.com/rmiyoyo/metrics-webapp/assets/120647236/9885dfce-1da9-4ba1-9107-1dd32c8a8a41

bellom commented:
Hi @rmiyoyo 

Great job so 🚀 👍.

This is a manual review

## Approved

rmiyoyo commented:
Thank you for your review @bellom!
